FixDevV3Blobs/FixDevV3Blobs/Program.cs

```diff
@@ -0,0 +1,192 @@
+            do
+            {
+                var segment = await container.ListBlobsSegmentedAsync(
+                    prefix: null,
+                    useFlatBlobListing: true,
+                    blobListingDetails: BlobListingDetails.None,
+                    maxResults: 5000,
+                    currentToken: continuationToken,
+                    options: null,
+                    operationContext: null);
+
+                var range = segment.Results.OfType<CloudBlockBlob>().ToList();
+                if (range.Count != segment.Results.Count())
+                {
+                    throw new Exception("Unexpected result item type: " + segment.Results.First(r => r.GetType() != typeof(CloudBlockBlob)).GetType());
+                }
+                blobList.AddRange(range);
+                continuationToken = segment.ContinuationToken;
+                Log("Discovered {0} blobs in {1}", blobList.Count, stopwatch.Elapsed);
+            } while (continuationToken != null);
+            stopwatch.Stop();
+            return blobList;
+        }
+
+        private static async Task<Stopwatch> ProcessBlobList(List<CloudBlockBlob> blobList, string search, string replace)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var tasks = Enumerable.Range(1, MaxTasks).Select(_ => Task.CompletedTask).ToArray();
+            var count = 0;
+            var skipped = 0;
+            foreach (var blob in blobList)
+            {
+                ++count;
+                //await ProcessBlob(blob);
+                if (Processed(blob))
+                {
+                    ++skipped;
+                    continue;
+                }
+                await StartProcessing(tasks, blob, search, replace);
+                if ((count % 1000) == 0)
+                {
+                    Log(
+                        "{0}/{1} (skipped: {4}) {2}, ETA: {3}",
+                        count,
+                        blobList.Count,
+                        stopwatch.Elapsed,
+                        TimeSpan.FromSeconds(((double)blobList.Count - count) / count * stopwatch.Elapsed.TotalSeconds),
+                        skipped);
+                }
+            }
+            await Task.WhenAll(tasks);
+            stopwatch.Stop();
+            return stopwatch;
+        }
+
+        private static bool Processed(CloudBlockBlob blob)
+        {
+            return ProcessedBlobs.Contains(blob.Uri.AbsoluteUri);
+        }
+
+        private static async Task StartProcessing(Task[] tasks, CloudBlockBlob blob, string search, string replace)
+        {
+            await Task.WhenAny(tasks);
+
+            for (int i = 0; i < tasks.Length; ++i)
+            {
+                if (tasks[i].IsCompleted)
+                {
+                    tasks[i] = ProcessBlob(blob, search, replace);
+                    return;
+                }
+            }
+
+            throw new Exception("Failed to find completed task");
```

Comment:
nit: please throw `InvalidOperationException`. using base exception is a bad practice since it makes the catch site very unclear and overly broad